### PR TITLE
[Balance] [QoL] Add Exclusive Moves from a Prior Evolution to Evolutions via Memory Mushroom

### PR DESCRIPTION
**Pull request by Blitz425**
## What are the changes the user will see?
Any Pokemon that had moves that were otherwise exclusive to a pre evolution now have those moves accessible through Memory Mushroom means (or through level for certain cases). 

## Why am I making these changes?
Requested by Damo and a common request in the discord, certain Pokemon such as Misdreavus or Pansage/Pansear/Panpour had their level up moveset neutered if you either:

A - Got lucky by finding an Evolution Stone and chose to evolve early on
or
B - Chose to forget a move that was exclusive from the pre evolution / Accidentally forgetting a move that was from the pre evolution.

This is mainly a QoL change for cases such as these, or for retaining cool, niche utility from any pre evolutions that the player can easily miss such as Sticky Web on certain Pokemon. There are also a few cases where a Pokemon miraculously loses a move in their 2nd stage such as Charmeleon losing access to Fire Spin, or Gothorita / Arctibax losing Psychic and Dragon Claw respectively. 

## What are the changes from a developer perspective?
Fixed Comments to be consistent across the board
Fixed the order of Poliwrath's moves
Addition of Moves for certain Pokemon in pokemon-level-moves.ts with comments signifying their inclusion / purpose.

<details>
  <summary>POKEMON AFFECTED </summary>

- Charmeleon Fire Spin (Level 1)
- Metapod  Tackle, String Shot, Bug Bite (Relearn Only)
- Kakuna - Poison Sting, String Shot, Bug Bite (Relearn Only)
- Beedrill - Harden, Poison Sting, String Shot, Bug Bite (Level 1)
- Sandslash - Agility (Level 1)
- Primeape / Annihilape - Covet (Level 1)
- Kadabra - Confusion (Level 1)
- Victreebel - Wrap, Growth, Acid, Knock Off, Poison Jab, Slam (Relearn Only), Poison Powder, Stun Spore (Level 1)
- Golem / Alola Golem - Rollout (Level 1)
- Haunter / Gengar - Acid (Level 1)
- Mr. Mime - Tickle (Level 1)
- Vaporeon/Jolteon/Flareon/Espeon/Umbreon/Leafeon/Glaceon/Sylveon - Veevee Volley (Relearn Only)
- Snorlax - Odor Sleuth (Level 1)
- Marill / Azumarill - Splash (Level 1)
- Sudowoodo - Tackle (Level 1)
- Politoed - Water Sport (Level 1)
- Sunflora - Seed Bomb (Relearn Only), Endeavor, Synthesis (Level 1)
- Slowing - Zen Headbutt (Level 24)
- Steelix - Gyro Ball (Level 22)
- Scizor - Steel Wing (Level 30)
- Ursaring / Ursaluna - Fling, Baby-Doll Eyes, Charm (Level 1)
- Kingdra - Laser Focus (Relearn Only)
- Donphan - Tackle, Odor Sleuth, Flail, Endure, Take Down, Charm, Last Resort, Double-Edge (Level 1)
- Porygon2 - Magic Coat (Level 1)
- Marshtomp - Surf (Relearn Only)
- Silcoon / Cascoon - Tackle, String Shot, Poison Sting, Bug Bite (Relearn Only)
- Beautifly / Dustox - Tackle (Level 1)
- Lombre - Rain Dance (Level 1)
- Ludicolo - Leech Seed, Giga Drain (Relearn Only)
- Nuzleaf - Bide, Headbutt (Level 1)
- Shiftry - Bide, Headbutt (Relearn Only)
- Masquerain - Baton Pass, Sticky Web (Relearn Only), Bubble Beam, Agility, Mist, Haze (Level 1)
- Breloom - Giga Drain (Level 1)
- Vigoroth - Play Rough (Relearn Only), Yawn, Slack Off, Headbutt, Amnesia, Covet, Flail (Level 1)
- Slaking - Play Rough, Focus Punch (Relearn Only), Focus Energy, Uproar, Fury Swipes, Endure, Headbutt, Slash, Reversal (Level 1)
- Hariyama - Sand Attack (Level 1)
- Swalot - Pain Split (Level 1)
- Wailord - Tackle (Level 1)
- Camerupt - Flamethrower, Double-Edge (Relearn Only)
- Grumpig - Growl (Level 1)
- Banette - Astonish, Pursuit (Level 1)
- Dusclops - Pursuit (Level 1)
- Chimecho - Last Resort, Entrainment (Relearn Only)
- Huntail / Gorebyss - Clamp (Level 1)
- Salamence - Outrage (Relearn Only)
- Grotle / Torterra - Growth (Level 1)
- Monferno / Infernape - Nasty Plot, Facade, Flamethrower (Relearn Only)
- Prinplup / Empoleon - Charm (Level 1)
- Wormadam (All Forms) / Mothim - Struggle Bug (Level 1)
- Floatzel - Tackle (Level 1)
- Lopunny - Frustration (Level 1)
- Mismagius - Confusion, Confuse Ray, Mean Look, Hex, Psybeam, Pain Split, Payback, Shadow Ball, Perish Song (Relearn Only)
- Honchkrow - Peck, Gust, Night Shade, Assurance, Taunt, Mean Look, Torment (Level 1)
- Skuntank - Acid Spray (Level 1)
- Lucario - Endure (Level 1)
- Lickilicky - Tackle (Level 1)
- Yanmega - Air Cutter, Wing Attack (Level 1)
- Gliscor - Poison Sting, Poison Tail, Slash (Level 1)
- Probopass - Harden, Rock Throw (Level 1)
- Dusknoir - Pursuit (Level 1)
- Froslass - Weather Ball (Level 1)
- Watchog - Work Up (Relearn Only)
- Simisage - Scratch, Play Nice, Vine Whip, Leech Seed, Bite, Torment, Fling, Acrobatics, Grass Knot, Recycle, Natural Gift, Crunch (Relearn Only)
- Simisear - Scratch, Play Nice, Incinerate, Yawn, Bite, Amnesia, Fling, Acrobatics, Fire Blast, Recycle, Natural Gift, Crunch (Relearn Only)
- Simipour - Scratch, Play Nice, Water Gun, Water Sport, Bite, Taunt, Fling, Acrobatics, Brine, Recycle, Natural Gift, Crunch (Relearn Only)
- Musharna - Psywave (Level 1)
- Leavanny - Sticky Web, Bug Buzz (Relearn Only), Protect, Grass Whistle, Endure, Flail (Level 1)
- Zoroark - Scary Face (Level 1)
- Gothorita - Psychic (Level 40)
- Vanillish / Vanilluxe - Powder Snow (Level 1)
- Eelektrik - Tackle (Level 1)
- Eelektross - Thunderbolt, Acid Spray (Relearn Only), Tackle, Thunder Wave, Spark, Charge Beam, Ion Deluge, Bind (Level 1)
- Braviary / Hisui Braviary - Brave Bird (Relearn Only)
- Greninja / Ash Greninja - Round, Fling, Smack Down, Bounce (Level 1)
- Spewpa - Tackle, String Shot, Stun Spore, Bug Bite (Relearn Only)
- Vivillon - Protect, Tackle, String Shot, Harden, Bug Bite (Level 1)
- Florges - Vine Whip, Tackle, Fairy Wind, Razor Leaf (Level 1)
- Hisui Sliggoo - Body Slam (Level 1)
- Goodra - Acid Armor (Level 1), Dragon Pulse (Level 35)
- H-Goodra - Acid Armor (Level 1)
- Gourgeist - Leafage (Level 1)
- Crabominable - Crabhammer (Relearn Only), Vise Grip, Payback (Level 1)
- Araquanid - Water Sport (Level 1)
- Steenee / Tsareena - Leafage (Level 1)
- Cosmoem / Solgaleo / Lunala - Splash, Stored Power (Level 1)
- Flapple / Appletun - Leafage (Level 1)
- Hydrapple - Leafage (Level 1), Dragon Cheer (Relearn Only)
- Sandaconda - Mud Slap (Level 1)
- Polteageist - Absorb (Level 1)
- Galar Mr. Mime / Mr. Rime - Tickle, Barrier (Level 1)
- Dragapult - Dragon Pulse (Relearn Only)
- Wyrdeer - Me First (Level 1)
- Kleavor - Wing Attack, Air Slash (Relearn Only), Steel Wing (Level 30)
- Basculegion - Final Gambit (Relearn Only)
- Garganacl - Iron Defense (Relearn Only), Smack Down (Level 1)
- Grafaiai - Bite (Level 1)
- Rabsca - Mud Shot, Dig (Level 1)
- Clodsire - Tackle (Level 1)
- Dudunsparce - Tackle (Level 1)
- Dipplin - Leafage (Level 1)
- Arctibax - Dragon Claw (Level 32)
- Sinistcha - Giga Drain (Relearn Only)
- Archaludon - Laser Focus (Relearn Only)
- Hisui Samurott - Soak (Level 1)
- Hisui Decidueye - Nasty Plot (Relearn Only), Astonish (Level 1)
- Hisui Lilligant - Charm (Level 1)

</details>

### Screenshots/Videos
Before
image

After
image

## How to test the changes?
Overrides

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by xsn34kzx — APPROVED** [review]

**Review by DayKev — DISMISSED** [review]

**Review by damocleas — APPROVED** [review]
lgtm-looks-good-to-me

**Comment by damocleas** [comment]
do not commit yet

File: src/data/balance/pokemon-level-moves.ts [hunk 1]
```diff
@@ -15339,6 +15600,7 @@ export const pokemonSpeciesLevelMoves: PokemonSpeciesLevelMoves = {
   [Species.WYRDEER]: [
     [ EVOLVE_MOVE, Moves.PSYSHIELD_BASH ],
     [ 1, Moves.TACKLE ],
+    [ 1, Moves.ME_FIRST ], //Stantler
```
Comment (innerthunder) [inline src/data/balance/pokemon-level-moves.ts:15603]:
```suggestion
    [ 1, Moves.ME_FIRST ], // Previous Stage Move
```

**Review by innerthunder — CHANGES_REQUESTED** [review]
I ran a script on `pokemon-level-moves` to get this list: [pre-evo-exclusive-moves.txt](https://github.com/user-attachments/files/17471746/pre-evo-exclusive-moves.txt)

Missing moves atm:
- Butterfree: Harden
- Dipplin: Leafage

**Comment by Blitz425** [comment]
> I ran a script on `pokemon-level-moves` to get this list: [pre-evo-exclusive-moves.txt](https://github.com/user-attachments/files/17471746/pre-evo-exclusive-moves.txt)
> 
> Missing moves atm:
> 
> * Butterfree: Harden
> * Dipplin: Leafage

Butterfree already had Harden in its learnset, I did miss Dipplin though. Thank you!

**Review by damocleas — APPROVED** [review]

**Review by DayKev — APPROVED** [review]

**Pull request merged**